### ISSUE TITLE
Incremental compialtion returns wrong positions for semanticdb

### ISSUE THREAD

**Issue by tgodzik** [lampepfl/dotty#10885]
## Minimized code

This is easy to see with code lenses in Metals - they change their position without any meaningful changes.

Let's say we compile

```Scala

def other = 123

@main def hello: Unit = {
  println("Hello world!")
}
```
this will give us at least a correct line for symbol:
`SymbolOccurrence(Some(Range(5, 10, 5, 15)), "_empty_/Main$package.hello().", DEFINITION)`

if we remove `other` method we get:
`SymbolOccurrence(Some(Range(2, 12, 2, 17)), "_empty_/Main$package.hello().", DEFINITION)`

which is nowhere near the actual line.

After adding a meaningless space we get the correct result again:

`SymbolOccurrence(Some(Range(4, 10, 4, 15)), "_empty_/Main$package.hello().", DEFINITION)`

## Expectation

Incremental compilation should give us the correct positions, otherwise we will be unable to support multiple features including `definition`, `references`, `rename` and `code lenses`

**Comment by bishabosha** [comment]
This issue only appears when using bloop for incremental compilation and not sbt 1.4.6

**Comment by tgodzik** [comment]
This seems to be fixed in 3.0.0-RC1 :tada:

**Comment by bishabosha** [comment]
I seem to be getting the same problem as before, at least how code lenses in metals behave

**Comment by megri** [comment]
@bishabosha Yes, the original issue (https://github.com/scalameta/metals/issues/2338) was reopened so this one should probably be as well. Or is the issue in bloop itself?

**Comment by tgodzik** [comment]
Let's reopen at least. It's more visible in the Dotty repo.
